test/helpers.rb

```diff
@@ -44,4 +45,17 @@ def filechecks(except = nil)
 
     checks
   end
+
+  def tempfile_with_data(filename, data)
+    file = Tempfile.new(filename)
+    file.write(data)
+    file.rewind
+
+    begin
+      yield file
+    ensure
+      file.close
+      file.unlink
+    end
+  end
```

Comment:
I think you can simplify this further to (the only difference being that the file won't be immediately deleted, see the `Tempfile` documentation):

```ruby
def tempfile_with_data(filename, data)
  Tempfile.open(filename) do |file|
    file.write(data)
    file.rewind
    yield file
  end
end
```